ha/k8s_setup/ha_setup.py

```diff
@@ -187,11 +188,21 @@ def process(self):
         Process config command.
         """
         try:
-            consul_endpoint = Conf.get(self._index, f'cortx{_DELIM}external{_DELIM}consul{_DELIM}endpoints[0]')
-            if not consul_endpoint:
-                sys.stderr.write(f'Failed to get consul config. consul_config: {consul_endpoint}. \n')
+            consul_endpoints = Conf.get(self._index, f'cortx{_DELIM}external{_DELIM}consul{_DELIM}endpoints')
+            #========================================================#
+            # consul Service endpoints from cluster.conf             #
+            #____________________ cluster.conf ______________________#
+            # endpoints:                                             #
+            # - tcp://consul-server.default.svc.cluster.local:8301   #
+            # - http://consul-server.default.svc.cluster.local:8500  #
+            #========================================================#
+            # search for supported consul endpoint url from list of configured consul endpoints
+            filtered_consul_endpoints = list(filter(lambda x: urlparse(x).scheme == const.consul_scheme, consul_endpoints))
+            if not filtered_consul_endpoints:
```

Comment:
I think the other check is also necessary. Because, for ex:
In case, if you get consul_endpoint = [''], in that case, that check is necessary.

Reply:
Discussed with Mukhtar. So, I think its not needed.

Reply:
------------------------------------------
sample endpoint entries of consul config from cluster.conf
------------------------------------------
...
external:
consul:
admin: admin
endpoints:
- tcp://consul-server.default.svc.cluster.local:8301
-
- http
- http:
- http://
- http://127.0.0.1:8500
- https:///consul-server.default.svc.cluster.local:8500
- http://consul-server.default.svc.cluster.local:8500
...
------------------------------------------



so ha_setup will get the consul endpoints
consul endpoints:
[
'tcp://consul-server.default.svc.cluster.local:8301',
None,
'http',
{'http': None},
'http://',
'http://127.0.0.1:8500',
'https:///consul-server.default.svc.cluster.local:8500',
'http://consul-server.default.svc.cluster.local:8500'
]



filert to below output:
filtered endpoints:
[
'http://',
'http://127.0.0.1:8500',
'http://consul-server.default.svc.cluster.local:8500'
]



and out of filter if used invalid entry will get error like below
HA Config failed. OS_error: Invalid URL '<url>': No host supplied.